x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameIndexer.java

```diff
@@ -226,6 +293,72 @@ protected boolean handleCircuitBreakingException(Exception e) {
         return true;
     }
 
+    private void getChangedBuckets(DataFrameTransformCheckpoint oldCheckpoint, DataFrameTransformCheckpoint newCheckpoint,
+            ActionListener<Map<String, List<String>>> listener) {
+        SearchRequest searchRequest = new SearchRequest(getConfig().getSource().getIndex());
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+
+        // we do not need the sub-aggs
+        CompositeAggregationBuilder changesAgg = pivot.buildChangedBucketsAggregation(pageSize);
```

Comment:
It seems to me that we also only care about the `terms` agg. If we keep the `histogram` and `date_histogram` in the composite agg, we could have tons of buckets to scroll through for no reason.

Reply:
for now yes, to be implemented later. I think for histogram this approach is fine, there aren't many anyways. `date_histogram` lacks an implementation.

Mixed sources (terms + (date_)histogram) shall be fine as well.